doc/interfaces/reconstruction_flow.rst

```diff
@@ -0,0 +1,269 @@
+.. _reconstruction_flow:
+
+==============
+Reconstruction
+==============
+
+This tutorial walks through the steps to perform reconstruction using DIPY.
+Multiple reconstruction methods are available in DIPY.
+
+You can try these methods using your own data; we will be using the data in DIPY.
+You can check how to :ref:`fetch the DIPY data<data_fetch>`.
+
+-----------------------------------
+Constrained Spherical Deconvolution
+-----------------------------------
+
+This method is mainly useful with datasets with gradient directions acquired on a
+spherical grid.
+
+The basic idea with this method is that if we could estimate the response function
+of a single fiber then we could deconvolve the measured signal and obtain the
+underlying fiber distribution.
+
+In this way, the reconstruction of the fiber orientation distribution function
+(fODF) in CSD involves two steps:
+
+* Estimation of the fiber response function
+* Use the response function to reconstruct the fODF
+
+We are using ``stanford_hardi`` dataset. You can also use your own data.
+
+Create the directories in which to save the peaks volume (e.g.: ``recons_csd_output``)
+and mask file (e.g.: ``stanford_hardi_mask``)::
+
+    mkdir recons_csd_output stanford_hardi_mask
+
+The workflow for the CSD reconstruction method requires the paths to the
+diffusion input file, b-values file, b-vectors file and mask file. The optional
+arguments can also be provided. In this case, we will specify FA threshold for
+calculating the response function (``fa_thr``), spherical harmonics order used
+in the CSA fit (``sh_order``), whether to use parallelization in peak-finding
+during the calibration procedure or not (``parallel``), and the output
+directory (``out_dir``).
+
+To get the mask file, we will run the following command::
+
+    dipy_median_otsu data/stanford_hardi/HARDI150.nii.gz --vol_idx 3 4 5 6 --out_dir "stanford_hardi_mask"
+
+Then, to perform the CSD reconstruction we will run the command as::
+
+    dipy_fit_csd data/stanford_hardi/HARDI150.nii.gz data/stanford_hardi/HARDI150.bval data/stanford_hardi/HARDI150.bvec stanford_hardi_mask/brain_mask.nii.gz --fa_thr 0.9 --sh_order 9 --parallel True --out_dir "recons_csd_output"
```

Comment:
@arokem the Stanford HARDI dataset has 160 gradient encoding directions. The maximum SH order could be much higher, right? Is there any reason not to use higher orders (besides the computational time)?

Reply:
@arokem can you please comment on this? Thanks !

Reply:
Sorry I missed this until now. 

It's true that in principle there are enough gradient encoding directions to fit a higher order of SH. The question is whether you think there is any information in the data to tell you about details at that level of angular resolution. 

An analogy that may (or may not...) be helpful is that you wouldn't bother modeling the spectrum of an fMRI BOLD signal with frequencies much higher than 0.1 Hz, even if you had data sampled at much higher sampling rate. That's because the data just doesn't have any useful information at these higher frequencies. 

So, I've never really use an SH order even this high in practice. I usually use something between 4 and 8. 

Does that answer your question? Does it require additional explanation in the text?

Reply:
Yes, it does answer the question. Thanks @arokem ! The rationale sounds reasonable.

So maybe then using 8 in order to be consistent with the DIPY [CSD reconstruction example](https://github.com/dipy/dipy/blob/master/doc/examples/reconst_csd.py) is a more common choice for this case.

A similar comment applies to the `fa_thr` parameter: a value 0.7 is consistent with the Python example.

Although those are also the default values for the CLI command, making them explicit here is reasonable to me.

Reply:
Addressed in 5a6b679